File: src/types/Restaurante/EmpleadoPrivilegio.ts
import { Privilegios } from "./Privilegios";

export class EmpleadoPrivilegio {
    id: number = 0;
    privilegio: Privilegios = new Privilegios(0, '', []);
    permisos: string[] = [];

    constructor(id: number, privilegio: Privilegios, permisos: string[]) {
        this.id = id;
        this.privilegio = privilegio;
        this.permisos = permisos;
    }
}

File: server/api/src/main/java/main/entities/Restaurante/EmpleadoPrivilegio.java
package main.entities.Restaurante;


import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class EmpleadoPrivilegio {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "empleado_id")
    private Empleado empleado;

    @ManyToOne
    @JoinColumn(name = "privilegio_id")
    private Privilegios privilegio;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "permisos_empleado", joinColumns = @JoinColumn(name = "empleado_privilegio_id"))
    @Column(name = "permiso")
    private List<String> permisos = new ArrayList<>();
}
